File: app/src/main/java/com/semisonfire/cloudgallery/utils/DateUtils.java
package com.semisonfire.cloudgallery.utils;


import android.text.format.DateFormat;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;
import java.util.TimeZone;

public class DateUtils {

    public static final String ONLY_DATE_FORMAT = "dd.MM.yyyy";
    public static final String DEFAULT_FORMAT = "yyyy-MM-dd'T'HH:mm:ss";

    public static String getCurrentDate() {
        long millis = System.currentTimeMillis();
        return getDateString(new Date(millis), DEFAULT_FORMAT);
    }

    public static String getDateString(String date, String format) {
        try {
            SimpleDateFormat simpleDateFormat = new SimpleDateFormat(DEFAULT_FORMAT, Locale.getDefault());
            simpleDateFormat.setTimeZone(TimeZone.getTimeZone("GMT"));
            Date newDate = simpleDateFormat.parse(date);
            return getDateString(newDate, format);
        } catch (ParseException e) {
            e.printStackTrace();
        }
        return null;
    }

    public static String getDateString(Date date, String format) {
        return DateFormat.format(format, date).toString();
    }
}

File: app/src/main/java/com/semisonfire/cloudgallery/utils/DateUtils.kt
package com.semisonfire.cloudgallery.utils

import android.text.format.DateFormat
import com.semisonfire.cloudgallery.logger.printThrowable
import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

object DateUtils {
    private const val DATE_FORMAT = "dd.MM.yyyy"
    private const val DATE_TIME_FORMAT = "yyyy-MM-dd'T'HH:mm:ss"

    private val dateFormat = SimpleDateFormat(DATE_FORMAT, Locale.ROOT)
    private val dateTimeFormat = SimpleDateFormat(DATE_TIME_FORMAT, Locale.ROOT)

    val currentDate: String
        get() {
            return millisToDateTime(System.currentTimeMillis())
        }

    fun dateString(date: String): String {
        return dateString(dateTimeFormat.parse(date), DATE_FORMAT)
    }

    fun dateTimeMillis(date: String?): Long {
        try {
            return dateTimeFormat.parse(date.orEmpty())?.time ?: -1L
        } catch (e: ParseException) {
            e.printThrowable()
        }

        return -1L
    }

    fun dateMillis(date: String?): Long {
        try {
            return dateFormat.parse(date.orEmpty())?.time ?: -1L
        } catch (e: ParseException) {
            e.printThrowable()
        }

        return -1L
    }

    fun millisToDate(millis: Long): String {
        return DateFormat.format(DATE_FORMAT, millis).toString()
    }

    fun millisToDateTime(millis: Long): String {
        return DateFormat.format(DATE_TIME_FORMAT, millis).toString()
    }

    private fun dateString(date: Date?, format: String?): String {
        return DateFormat.format(format, date).toString()
    }
}